osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/InvocationCounterManagerTest.java

```diff
@@ -93,4 +101,41 @@ void initializesInvocationCounterForDevice() throws Exception {
     assertThat(this.device.getVersion()).isGreaterThan(this.initialDeviceVersion);
     assertThat(this.device.getInvocationCounter()).isEqualTo(invocationCounterValueOnDevice);
   }
+
+  @Test
+  void attemptToLowerInvocationCounterForDeviceLogsErrorAndThrowsException() throws Exception {
+    final Logger invocationManagerLogger =
+        (Logger) LoggerFactory.getLogger(InvocationCounterManager.class);
+    final ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+    final List<ILoggingEvent> logsList = listAppender.list;
+    final long invocationCounterValueOnDevice = 1;
```

Comment:
I'm not sure what you mean here but I do see that it can be made more clearer. I have tried to improve it by giving them better names, so it is more self explainable.

Reply:
The new name provides information about the invocation counter on the device.
This could help clarifying the expectation should the test fail.
With something like the following, the relevant values for the invocation counter are visible and controlled local to this single test:
```
    final long invocationCounterValueOnDevice = 1;
    final long previouslyKnownInvocationCounter = 7; // bigger than invocationCounterValueOnDevice
    this.device.setInvocationCounter(previouslyKnownInvocationCounter);
```
An alternative could be a builder method for a device where the invocation counter is a parameter, and use that to obtain the device.

Another advantage would be that somebody who makes changes to the general before-each set up of the device would not have to take into account which values are needed by which tests.
If for some other test a value of 83 would come in more convenient, this test might still succeed, but `deviceWithPreviouslyKnownInvocationCounterSeven` would no longer makes sense. This too can be avoided by taking the expected value (7) into the test implementation instead of referring to it inside the general `setUp` by naming the device variable.